src/components/TilePicker.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import * as db from "../firebase";
+import { useForm } from "react-hook-form";
+
+export function TilePicker(props: {
+  tileName: string,
+  tileInfo: db.TileInfo,
+  map: db.MapRepr,
+  tiles: db.TileDict,
+  selectedRow?: number,
+  selectedCol?: number,
+  gameId: string
+}) {
+  const { register, handleSubmit } = useForm<{tileName: string}>();
+
+  const tileList = [...Object.keys(props.tiles)]
+  const tiles = tileList.map( (tileName) =>
+    <option key={tileName}>{tileName}</option>
+  )
+
+  const onSubmit = handleSubmit(async ({tileName}) => {
+    changeTile(tileName);
+  });
+
+  function changeTile(tileName: string) {
+    if (!tileList.includes(tileName)) {
+      createTile(tileName);
+    } else {
+      const row = props.selectedRow;
+      const col = props.selectedCol;
+      if (row !== undefined && col !== undefined) {
+        props.map[row][col] = tileName;
```

Comment:
I think directly modifying a state/prop variable is frowned upon? Can we copy it?